File: 350. Intersection of Two Arrays II/intersectionoftwoarrays2.java
import java.util.Arrays;

class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        int[] seen = new int[1001];
        int[] result = new int[1001];
        int size = 0;

        for (int num : nums1) {
            seen[num]++;
        }

        for (int num : nums2) {
            if (seen[num] > 0) {
                result[size++] = num;
                seen[num]--;
            }
        }

        return Arrays.copyOf(result, size);
    }
}

File: 350. Intersection of Two Arrays II/intersectionoftwoarrays2.py
class Solution:
    def intersect(self, nums1: list[int], nums2: list[int]) -> list[int]:
        seen = [0] * 1001
        result = []

        for num in nums1:
            seen[num] += 1

        for num in nums2:
            if seen[num] > 0:
                result.append(num)
                seen[num] -= 1

        return result
